### DISCUSSION TITLE
[Feature Request] Resize tab size

### DISCUSSION THREAD

**Discussion by TroySigX** [ghostty-org/ghostty#3297] [Feature Requests, Ideas] [answered]
Can we somehow resize the tab's height?

**Comment by jjt** [comment]
literally came to the ghostty repo to file this request 😄 

full config

```
font-family = "Monaspace Neon"
font-size = 14
adjust-cell-width = -15%
adjust-cell-height = -15%
font-thicken = false
theme = catppuccin-mocha
macos-titlebar-style = tabs
```

<img width="748" alt="Image" src="https://github.com/user-attachments/assets/b627cd2d-1678-431a-95f8-b1ad38c2c378" />

**Reply by ttys3** [reply]
> literally came to the ghostty repo to file this request 😄
> 
> full config
> 
> ```
> font-family = "Monaspace Neon"
> font-size = 14
> adjust-cell-width = -15%
> adjust-cell-height = -15%
> font-thicken = false
> theme = catppuccin-mocha
> macos-titlebar-style = tabs
> ```
> 

I do not think it is the correct way to set `adjust-cell-height` to decrease only tabs height.

because it is global, and I do not want to touch it.  I only want the tabs be more slim, not the whole terminal.

currently I use [custom css](https://github.com/ghostty-org/ghostty/discussions/3983) to do this

**Reply by jjt** [reply]
[`adjust-cell-height`](https://ghostty.org/docs/config/reference#adjust-cell-height) should have no effect on the tab height, but it's part of my full config and i didn't want to omit any context just in case there's unintended interactions somehow (v doubtful but still)

**Comment by rhodes-b** [comment]
@TroySigX the preferred method is to open a discussion first there is an option for new ideas

**Comment by rhodes-b** [comment]
It can then be converted to an issue when ready
![Image](https://github.com/user-attachments/assets/26d265a2-b765-4df0-8a5a-fcf95804c19b)

**Comment by xaviduds** [comment]
Also would like some config for that, its too thick:
![screenshot](https://github.com/user-attachments/assets/314d07e3-e499-4607-8b98-391aa5d94c56)
It could be based on font size, like rem, or have a % of its default value, like 0.65
example:
tab-height = 0.65
or
tab-height = (number)rem (1rem)

**Comment by xzbdmw** [comment]
I feel like neovide's titlebar height is more seamless.
<img width="1500" alt="image" src="https://github.com/user-attachments/assets/3396bbfd-6a38-443b-9d63-ed5fa048f2cb" />

<img width="1495" alt="image" src="https://github.com/user-attachments/assets/2b0c07c4-0c55-4cd5-a5d9-cca6d9bc100f" />

**Reply by ibehnam** [reply]
that's also how iTerm2 does it.

**Comment by pro-sumer** [comment]
I prefer iTerm's smaller tab height and hope we can configure Ghostty to be more like that some day.

<img src="https://github.com/user-attachments/assets/a3cfe73c-5914-4276-a1f1-e4e0a8431f42" alt="Ghostty" width=50%><img src="https://github.com/user-attachments/assets/a6439b9f-47f1-4deb-b8fe-d1be2e109f15" alt="iTerm" width=50%>

**Comment by sumgr0** [comment]
Also allow for the tabs list positioning to be configurable: top, bottom

**Reply by tamwile** [reply]
> Also allow for the tabs list positioning to be configurable: top, bottom

If you are using the GTK renderer, you can already use `gtk-tabs-location = bottom`
https://ghostty.org/docs/config/reference#gtk-tabs-location

**Comment by 1Mochiyuki** [comment]
![image](https://github.com/user-attachments/assets/87ce637c-7d77-4b1a-be22-f39524ff840c)
would personally love to see something similar to wezterm's tab styling or at least the ability to get something similar to it and top/bottom position config

**Reply by jjt** [reply]
this seems similar to kitty's tabs (screenshot below of my setup), which i like

taking this a step further my ideal state would be top to bottom terminal lines with no OS chrome at all, but still use tabs under the hood, including all `____tab` actions

more specifically i'd like:
1. option to float the tabs bar over top of the terminal lines as to not take up space
2. option to hide by default as to not obscure lines
3. actions such as `show_tabs_bar`, `hide_tabs_bar` and `toggle_tabs_bar`
4. if hidden, option to show when changing tabs and hide again after x seconds

i don't necessarily need it in the terminal style text, but taking up no more than one vertical line is important to me especially if it can't be toggled

my kitty setup (showing nvim editing ghostty's config):
<img width="756" alt="image" src="https://github.com/user-attachments/assets/46c5eda3-bbac-4724-aee4-231b65c038da" />

**Comment by horseinthesky** [comment]
Tiny accurate tabs would be so nice to have.
![image](https://github.com/user-attachments/assets/1781a241-b1ee-4017-a59f-c01d599bca0a)

**Comment by ttys3** [comment]
for Linux, this is already supported.  see https://github.com/ttys3/my-ghostty-config

![image](https://github.com/user-attachments/assets/d8c2b626-4500-4c93-91c6-146af0dc5d49)


use `gtk-custom-css` config option, which been impl in https://github.com/ghostty-org/ghostty/pull/4200

**Comment by shakedhagag** [comment]
Any updates on this? is it planned?

**Comment by rhodes-b** [comment]
Note for others on Linux use custom gtk css like suggested above

This request should now be limited to MacOS

**Comment by ibehnam** [comment]
Any update on this? The tab bar is too thick. I like how iTerm-2 does it:
<img width="640" alt="image" src="https://github.com/user-attachments/assets/239e5a3d-cc76-492a-a159-c3c095f3ae1d" />

**Comment by 00-kat** [comment]
Has anyone even determined if this is possible to do without switching to a custom tab implementation[^1]?  If it's not possible, then this will likely be some ways out.

[^1]: Apps like iTerm2 and Safari don't use the native tabs; they draw their own very similar-looking ones that are considerably less buggier than the native tabs macOS gives.

**Reply by lowstar** [reply]
Isn't ghostty already using custom tabs implementation when `macos-titlebar-style` is set to `tabs`?

**Reply by 00-kat** [reply]
I'm not sure, but based on my reading of the documentation I believe the *title bar* is custom but the tabs are not.  I've not heard of anyone knowledgeable in this say that custom tabs exist though.

**Comment by DerPlayer2001** [comment]
Is there any way one could help getting this FR started? Would rly like to see this!

**Comment by sampl3x** [comment]
Any updates regarding this option on macos?

**Comment by mitchellh** [comment, marked as answer]
Thanks everyone. I'm going to close this request. The point is received. All of the examples of reduced tab sizes given here are using custom tab bars. We're using the macOS native tab bar (even `titlebar-style = tabs` is just modifying the native tab bar). We're limited in what we can achieve. On that, a couple notes.

First, long term, we plan to implement a custom tab bar, so this will be resolved. We need to do this for a variety of reasons, but most notably so the the quick terminal and non-native fullscreen can support tabs. When doing so, we'll have full control over tab styling and make them thinner. 

Second, on macOS tahoe, the titlebar tabs are significantly thinner just thanks to Apple's design changes. I've heard happy reports from people on this. 

All in all, I'm closing this because it's going to happen one day no matter what, and saying "me too" isn't helping anymore. Thanks everyone.
